proto/regen/data/v1alpha1/query.proto

```diff
@@ -32,4 +36,25 @@ message QueryByCidResponse {
 
     // content is the content of the data, if it was stored on-chain.
     bytes content = 3;
-}
\ No newline at end of file
+}
+
+
+// QueryBySignerRequest is the Query/BySigner request type.
+message QueryBySignerRequest {
+
+    // signer is the address of the signer to query by.
+    string signer = 1;
+
+    // pagination is the PageRequest to use for pagination.
+    cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
```

Comment:
NOTE: It's good to have it for verification, but in most of the production use-cases we will probably like to limit the blockchain load, and use off-chain DB for queries and analytics.